### ISSUE TITLE
How to disable tests in ios_unit_test?

### ISSUE THREAD

**Issue by Tony1023** [bazelbuild/rules_apple#518]
In Xcode, we can disable tests in the file inspector. Those are recorded in xcscheme. Is there a way to do so with ios_unit_test?

**Comment by keith** [comment]
This isn't currently supported. We normally prefix the test name with `disabled_`

**Comment by Tony1023** [comment]
@keith Thanks, but will this be supported at some point?

**Comment by keith** [comment]
I haven't heard any plans, but someone from Google would have to answer that one.

**Comment by Tony1023** [comment]
@keith Thanks for the info! :)

**Comment by keith** [comment]
IMO you can keep this issue open until there's input from someone from Google, since this might be a reasonable feature request

**Comment by Tony1023** [comment]
@keith Yeah you're right. Will post in Bazel's Google groups too. If there are any updates from them I will update this issue too.

**Comment by thomasvl** [comment]
Are you looking to disable the whole test or just a test method?

In stock Xcode you don't really "disable" a test, as much as schemes control what tests/test methods the scheme invokes, if you create a new scheme, all of the tests are still fair game as nothing marks them as disabled in the code.

**Comment by allevato** [comment]
The closest thing to this is Bazel's [`--test_filter` command line flag](https://docs.bazel.build/versions/0.28.0/command-line-reference.html#flag--test_filter), but I believe it would be the responsibility of each test runner implementation to honor that setting and pass it through the underlying `xctest` command.

**Comment by Tony1023** [comment]
> Are you looking to disable the whole test or just a test method?

@thomasvl A whole test, but the test class is inherited by others so it's hard to remove it from the dependency list.

> 
> In stock Xcode you don't really "disable" a test, as much as schemes control what tests/test methods the scheme invokes, if you create a new scheme, all of the tests are still fair game as nothing marks them as disabled in the code.

Yes. So is there an easy way to let feed the xcscheme file to `ios_unit_test`?

**Comment by sergiocampama** [comment]
not really, we don't support xcschemes for running nor testing.

**Comment by louwers** [comment]
Do I understand it correctly that `test_filter` is a no-op for XCTest?

**Comment by keith** [comment]
`--test_filter` is supported as of the more recent releases. But that's really more useful for a single one off test run where you want to disable a test, not disabling permanently like that in Xcode.

**Comment by aaronsky** [comment]
There's also the `test_filter` attr, which should be respected by all of our test rules.
